x/foundation/keeper/internal/genesis_test.go

```diff
@@ -285,3 +288,58 @@ func TestImportExportGenesis(t *testing.T) {
 		require.Equal(t, tc.export, actual, name)
 	}
 }
+
+func TestShouldPanicWhenFailToGenerateFoundationModuleAccountInInitGenesis(t *testing.T) {
+	checkTx := false
+	app := simapp.Setup(checkTx)
+	testdata.RegisterInterfaces(app.InterfaceRegistry())
+	testdata.RegisterMsgServer(app.MsgServiceRouter(), testdata.MsgServerImpl{})
+	gs := &foundation.GenesisState{
+		Params:     foundation.DefaultParams(),
+		Foundation: foundation.DefaultFoundation(),
+	}
+	ctx := app.BaseApp.NewContext(checkTx, tmproto.Header{})
+
+	testCases := map[string]struct {
+		mockAccKeeper *stubAccKeeper
+	}{
+		"failed to generate foundation module account=" + foundation.ModuleName: {
```

Comment:
You mean, foundation.Treasury? Because the stub contains foundation.ModuleName.

Reply:
Yes, is it confusing?

Reply:
It says, failed to generate foundation.ModuleName, but it's opposite. Isn't it?

Reply:
Aha, My mistake.

Reply:
Changing the name of cases, or the logic in the stub would be required.